MonstieWash/Assets/Scripts/Audio/MonsterSoundManager.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(SoundPlayer))]
+public class MonsterSoundManager : MonoBehaviour
+{
+    [System.Serializable]
+    private struct MoodSound
+    {
+        public MoodType mood;
+        public Sound sound;
+    }
+
+    [SerializeField] private List<MoodSound> moodSounds = new();
+    [SerializeField] private List<Sound> attackSounds = new();
+
+    private SoundPlayer m_soundPlayer;
+    private MonsterBrain m_monsterBrain;
+    private MusicManager m_musicManager;
+
+    private void Awake()
+    {
+        m_soundPlayer = GetComponent<SoundPlayer>();
+        m_monsterBrain = GetComponent<MonsterBrain>();
+        m_musicManager = FindFirstObjectByType<MusicManager>();
+    }
+
+    private void OnEnable()
+    {
+        m_monsterBrain.OnMoodChanged += PlayMoodSound;
+        m_monsterBrain.OnMoodChanged += UpdateMusic;
+    }
+
+    private void OnDisable()
+    {
+        m_monsterBrain.OnMoodChanged -= PlayMoodSound;
+        m_monsterBrain.OnMoodChanged -= UpdateMusic;
+    }
+
+    /// <summary>
+    /// Plays a random attack sound. Will be updated when attacks are implemented.
+    /// </summary>
+    public void PlayAttackSound()
+    {
+        var index = Random.Range(0, attackSounds.Count - 1);
```

Comment:
Unity Random.Range actually works differently depending on whether it's returning a float or an int. For ints, max is exclusive, so this line would never return the final attacksound in the list.